src/singletons/NativeMessaging.cpp

```diff
@@ -60,6 +60,18 @@ void registerNmHost(const Paths &paths)
         auto obj = getBaseDocument();
         QJsonArray allowedOriginsArr = {
             u"chrome-extension://%1/"_s.arg(EXTENSION_ID)};
+
+        // Parse additional extension IDs from settings
+        QString additionalIDs = getSettings()->additionalExtensionIDs.getValue();
```

Comment:
This needs a `#include "singletons/Settings.hpp"` at the top (see failing builds).